src/components/Chat/Chat.tsx

```diff
@@ -244,7 +244,7 @@ const ChatWithLLM: React.FC<ChatWithLLMProps> = ({ currentFilePath }) => {
           )}
         </div>
         <div className="pr-2 pt-1 cursor-pointer" onClick={restartSession}>
-          <FiRefreshCw className="text-gray-300" /> {/* Icon */}
+          <FiRefreshCw className="text-gray-300" title="Restart Session" /> {/* Icon */}
```

Comment:
From the perspective of a user, this `Restart Session` button appears to do nothing. If the user is waiting on an response from their LLM, it aborts.

Reply:
Perhaps we could do something along the lines of setting a loader (and loading state) between the start and the end of restart session?